RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -222,36 +227,45 @@ public struct AccountRecoveryScanInProgress: Sendable, FeatureReducer {
 }
 
 extension AccountRecoveryScanInProgress {
-	private func derivePublicKeys(
-		state: inout State
-	) -> Effect<Action> {
+	private func nextDerivationPaths(state: inout State) throws -> OrderedSet<DerivationPath> {
 		let networkID = state.networkID
-		let used = state.indicesOfAlreadyUsedEntities
 
 		let derivationIndices = generateIntegers(
 			start: state.maxIndex ?? 0,
 			count: batchSize,
-			shouldInclude: { !used.contains($0) }
+			excluding: state.indicesOfAlreadyUsedEntities
 		)
 
 		assert(derivationIndices.count == batchSize)
 		state.maxIndex = derivationIndices.max()! + 1
 
-		let derivationPaths = try! OrderedSet(validating: derivationIndices.map {
+		return try OrderedSet(validating: derivationIndices.map {
```

Comment:
Style question: I would first calculate the array and only wrap it when returning it but it's up to you